objects/aidBoard/script.lua

```diff
@@ -849,21 +874,19 @@ function toggleElements()
 end
 
 function scanElements()
-    local elements = {0, 0, 0, 0, 0, 0, 0, 0}
-    local elementTokens = {0, 0, 0, 0, 0, 0, 0, 0}
+    local elements = Elements:new()
     for _, selected in pairs(Global.getVar("selectedColors")) do
-        for i=1,8 do
-            elements[i] = elements[i] + selected.elements[i].getButtons()[1].label
-            elementTokens[i] = elementTokens[i] + selected.elementTokens[i]
-        end
+        elements:add(selected.nonTokenElements)
     end
 
     local elementsTable = {"Sun","Moon","Fire","Air","Water","Earth","Plant","Animal"}
     for i,total in ipairs(elements) do
         local elementTokensCount = #getObjectsWithTag(elementsTable[i])
```

Comment:
i feel like it'd be more efficient to iterate through this list and check if the object has `Invocation Element` tag than to query all objects again

Reply:
My guess would be that it wouldn't, given that this code iterates in compiled code, where as your suggestion iterates in interpreted code.